modules/ItemManager/index.ts

```diff
@@ -0,0 +1,199 @@
+
+  constructor() {
+    this.selectedSection = null;
+    this.currentListId = null;
+  }
+
+  setSelectedSection(section: string | null) {
+    this.selectedSection = section;
+  }
+
+  setListId(listId: string | null) {
+    this.currentListId = listId;
+  }
+
+  /** applies or clears the selected store section */
+  filterBySection = (section: string | null) => {
+    this.setSelectedSection(this.selectedSection === section ? null : section);
+    this.populateItems();
+  };
+
+  /** Populate items in a specific list */
+  async populateItems() {
+    // in reality this function should be able to take 2 paths - drilled down for the list, or for ALL items
+    // which we might support viewing? I'm not sure what the use case would be...
+    const listId = this.currentListId;
+    if (!listId) return;
+    const list = await db.lists.where('id').equals(listId).toArray();
+
+    if (!list.length) return noListMessage();
+
+    const allListItems = await databaseService.getAllListItems(listId);
+
+    if (!allListItems.length) {
+      noItemsMessage();
+      return;
+    }
+
+    let sortedItems = filterItems(
+      allListItems,
+      hideChecked,
+      this.selectedSection
+    );
+
+    // arrange items based on auto sort or section sort
+    if (sectionSort && sectionSort.checked) {
+      sortedItems = sortItemsBySection(sortedItems);
+      // if section sort and auto sort is also turned on
+      if (autoSort && autoSort.checked) {
+        sortedItems = sortItemsByPurchaseStatus(sortedItems);
+      }
+    } else if (autoSort && autoSort.checked) {
+      sortedItems = sortItemsByPurchaseStatus(sortedItems);
+    }
+
+    const storeSectionsAndCounts = await generateStoreSectionData(allListItems);
+
+    renderItemsList(sortedItems);
+    renderSectionBubbles(storeSectionsAndCounts, this.selectedSection);
+  }
+
+  /** Add an item with/without a list */
+  async addItem(
+    name: string,
+    quantity: number,
+    quantityUnit: string,
+    price: number = 0,
+    section: string
+  ) {
+    await databaseService.addItem(
+      { name, quantity, quantityUnit, price, section },
+      this.currentListId
+    );
+    await this.populateItems();
+  }
+
+  async removeItem(id: string) {
+    // resetting selected section if we remove the last item in a section
+    // prevents user getting stuck in a selected section w/ no items
+    if (this.selectedSection) {
+      const itemToRm = await db.items.get(id);
+      const isOtherSection =
+        this.selectedSection === 'Other' && !itemToRm?.section;
+      const isSelectedSection = itemToRm?.section === this.selectedSection;
+
+      const itemsInSection = await countItemsInSection(isOtherSection, id);
+
+      const isItemToRmSingle =
+        (isOtherSection || isSelectedSection) && itemsInSection === 1;
+
+      if (isItemToRmSingle) {
+        this.setSelectedSection(null);
+      }
+    }
+
+    await databaseService.removeItem(id);
+
+    await this.populateItems();
```

Comment:
Instead of calling populateItems in here like this, we should just leave that to the caller, so we can further split out methods without relying on the inner function.